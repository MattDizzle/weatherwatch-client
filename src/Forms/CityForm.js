import React, { Component } from 'react';
import { CurrentContext } from '../CurrentContext';
import ValidationError from '../ValidationError';


export default class CityForm extends Component {
    static contextType = CurrentContext;

    validateCity() {
        const city = this.context.city.value.trim();
        if (city.length === 0) {
          return "City is required";
        } else if (city.length < 3) {
          return "City must be at least 3 characters long";
        }
      }
    
      validateProvince() {
        const province = this.context.province.value.trim();
        if (province.length === 0) {
          return "State/Province Invalid";
        } else if (province.length < 3) {
          return "State/Province must be at least 3 characters long";
        }
      }

    render() {
        const { getDataByCityAndState, updateCity, updateProvince, city, province, toggleLoadingTrue } = this.context;
        const cityError = this.validateCity();
        const provinceError = this.validateProvince();
        return (
            <form onSubmit={toggleLoadingTrue, getDataByCityAndState}>
              <label htmlFor="city">Enter a City </label>
              <input
                id="city"
                name="city"
                type="text"
                required
                onChange={(e) => updateCity(e.target.value)}
              />
              {city.touched && (
                <ValidationError message={cityError} />
              )}
              <label htmlFor="prov">and State/Province</label>
              <input
                id="prov"
                name="prov"
                type="text"
                onChange={(e) => updateProvince(e.target.value)}
              />
              {province.touched && (
                <ValidationError message={provinceError} />
              )}
              <button
                type="submit"
                className="btn"
                disabled={this.validateCity()}
              >
                <span>Submit</span>
              </button>
            </form>
        )
    }
}
