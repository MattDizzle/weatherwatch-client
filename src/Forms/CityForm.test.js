import React from "react";
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import CityForm from "./CityForm";
import CurrentContextProvider from "../CurrentContext";


describe(`CityForm Component`, () => {
    describe(`Smoke test`, () => {
        it("should render without crashing", () => {
            const div = document.createElement('div');
            ReactDOM.render(
            <CurrentContextProvider>
            <CityForm/>
            </CurrentContextProvider>,
             div);
            ReactDOM.unmountComponentAtNode(div);
          });
    })

    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
          const div = document.createElement('div');
            const tree = renderer.create(
            <CurrentContextProvider>
            <CityForm/>
            </CurrentContextProvider>
            ).toJSON()
            expect(tree).toMatchSnapshot()
        });
    });

})


