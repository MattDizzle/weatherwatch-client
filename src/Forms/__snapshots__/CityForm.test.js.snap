// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityForm Component Snapshot test Renders the UI as expected 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="city"
  >
    Enter a City 
  </label>
  <input
    id="city"
    name="city"
    onChange={[Function]}
    required={true}
    type="text"
  />
  <label
    htmlFor="prov"
  >
    and State/Province
  </label>
  <input
    id="prov"
    name="prov"
    onChange={[Function]}
    type="text"
  />
  <button
    className="btn"
    disabled="City is required"
    type="submit"
  >
    <span>
      Submit
    </span>
  </button>
</form>
`;
